mantidimaging/gui/dialogs/async_task/presenter.py

```diff
@@ -74,9 +77,12 @@ def progress_update(self) -> None:
         extra_info = progress_info[-1].extra_info
         self.progress_updated.emit(self.progress.completion(), msg if msg is not None else '')
 
-        if extra_info:
+        if extra_info and 'losses' in extra_info:
```

Comment:
if you agree, i think this may read nicer as:
```python
if extra_info:
    if 'losses' in extra_info:
        self.update_progress_plot(extra_info['iterations'], extra_info['losses'])
    if 'residual' in extra_info:
        self.progress_residual_plot_updated.emit(extra_info['residual'])
```